lib/charms/mongodb/v1/shards_interface.py

```diff
@@ -272,8 +269,40 @@ def update_mongos_hosts(self):
             self._update_relation_data(relation.id, {HOSTS_KEY: json.dumps(self.charm._unit_ips)})
 
     def get_config_server_status(self):
-        """TODO: Implement this function in a separate PR."""
-        return None
+        """Returns the current status of the config-server."""
+        if not self.charm.is_role(Config.Role.CONFIG_SERVER):
+            logger.info("skipping status check, charm is not running as a shard")
+            return None
+
+        if not self.charm.db_initialised:
+            logger.info("No status for shard to report, waiting for db to be initialised.")
+            return None
+
+        if self.model.relations[LEGACY_REL_NAME]:
+            return BlockedStatus(f"relation {LEGACY_REL_NAME} to shard not supported.")
+
+        if self.model.relations[REL_NAME]:
+            return BlockedStatus(f"relation {REL_NAME} to shard not supported.")
+
+        if not self.is_mongos_running():
+            return BlockedStatus("Internal mongos is not running.")
```

Comment:
Blocked is when we wait for action from the user. Maybe the Error status can be better.

Reply:
great idea

Reply:
can we fix this line?